internal/databroker/config_source.go

```diff
@@ -84,40 +89,50 @@ func (src *ConfigSource) rebuild(firstTime bool) {
 	var additionalPolicies []config.Policy
 
 	// add all the config policies to the list
-	for _, cfgpb := range src.dbConfigs {
+	for id, cfgpb := range src.dbConfigs {
 		cfg.Options.ApplySettings(cfgpb.Settings)
+		var errCount uint64
 
 		err := cfg.Options.Validate()
 		if err != nil {
-			log.Warn().Err(err).Msg("databroker: invalid config detected, ignoring")
-			return
+			metrics.SetDBConfigRejected(cfg.Options.Services, id, cfgpb.version, err)
+			continue
```

Comment:
do we not want to fail fast in this case?

Reply:
Could this cause unexpected issues since we typically fail hard after failed validation?

Reply:
it loops over available databroker configs. idk what other configs were originally planned to exist besides one supplied by the console, but as long as we're looping over and merging them, it should be more natural to just proceed to next one.

Reply:
by returning we are ignoring other configs that may be valid. your logic is valid in case we have just a single config, however by design we have multiple here that we merge. 

@calebdoxsey please advise

Reply:
also looping over map is random order that makes it even more non-deterministic. 
maybe we should eliminate the multiple databroker configs altogether here

Reply:
I don't know what to do if config is invalid.

Reply:
Was discussing with @wasaga - 

My understanding of the current code is that we're build a new config from the databroker by starting with a copy of current state, then mutating this copy.  It looks like we abort if the overall config doesn't validate, though we continue if individual routes have issues.  Is that right?

What I think I'd _want_ is to abort making changes if the new configuration is invalid for any reason.  If we proceed through a partial validation error, a previously functioning route or option could get dropped and cause an outage or otherwise undesirable state.  Preventing me from making new changes would retain whatever the current (likely functional) state is.  This does block new changes from happening, but I think that's the least surprising result and matches `HUP` behavior in a lot of other software.

Reply:
ok I'm reverting to original behaviour